src/app/services/focus.service.ts

```diff
@@ -17,8 +17,18 @@ export class FocusService {
 
     restoreFocus(): void {
       if (this.lastFocusedElement) {
-        this.lastFocusedElement.focus();
-        this.lastFocusedElement = null;
+        setTimeout(() => {
+          const dataTestValue = this.lastFocusedElement.getAttribute('data-test');
+
+          if (dataTestValue) {
+            const dataTestElement = this.document.querySelector(`[data-test="${dataTestValue}"]`);
+            (dataTestElement as HTMLElement)?.focus();
+          } else {
+            this.lastFocusedElement?.focus();
+          }
+
+          this.lastFocusedElement = null;
+        }, 200);
```

Comment:
Is it related somehow to animation duration?

Reply:
> Is it related somehow to animation duration?

possibly, it's just timing when element will be available after table re-rendered.